### PR TITLE
Fix coverity issues. Skip tests involving managed memory when not supported.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michał Zientkiewicz <mzient@gmail.com>

## Description
- [x] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR

Fixed bugs:
- some uninitialized variables/fields
- resource leaks in tests in case when the test fails catastrophically
- use of uninitialized in case of test failure
Other:
- skip tests that use managed memory when an allocation attempt raises an error.

#### Additional information
- Affected modules and functionalities:
* tests
* slice_kernel_utils
* stft setup
* copy kernel

- Key points relevant for the review:

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2208

### CODE REVIEW

File: dali/pipeline/util/copy_with_stride_test.cc [hunk 1]
```diff
@@ -78,11 +79,10 @@ TEST(CopyWithStrideTest, TwoDimsGPU) {
                      41, 42, 43, 44};
   Index stride[] = {8 * sizeof(size_t), sizeof(size_t)};
   Index shape[] = {2, 4};
-  size_t *data;
-  CUDA_CALL(cudaMalloc(reinterpret_cast<void**>(&data), sizeof(size_t) * 16));
+  DeviceBuffer<size_t> data, out;
+  data.from_host(h_data);
```
Comment (JanuszL) [inline dali/pipeline/util/copy_with_stride_test.cc:83]: Do you need `cudaMemcpy` if you call `from_host`?

Comment (mzient) [reply dali/pipeline/util/copy_with_stride_test.cc:83]: No, I don't.

**Review by JanuszL — APPROVED** [review]

**Review by awolant — APPROVED** [review]
I think that in the future we should avoid doing two unrelated things in one PR, if possible.

**Comment by mzient** [comment]
> I think that in the future we should avoid doing two unrelated things in one PR, if possible.

Hopefully, some day we'll have CI that runs in 10 minutes (maaaybe 20) and then... you'd see how much more willing we'd all be to stick to separation of concerns....

**Pull request merged**